modules/vocabularies/src/main/java/ca/sickkids/ccm/lfs/vocabularies/VocabularyTermInfoServlet.java

```diff
@@ -0,0 +1,233 @@
+                vocabTermNodeName = vocabTermNodeName.replaceAll(":", "");
+
+                Resource linkedValue = resolver.getResource(vocab, "./" + vocabTermNodeName);
+                if (linkedValue == null) {
+                    // We couldn't load the parent resource; leave it as is
+                    builder.add(value);
+                } else {
+                    JsonObject linkedObject = linkedValue.adaptTo(JsonObject.class);
+                    builder.add(populateChildren(linkedObject, resolver, parentpath));
+                }
+            } else {
+                // This is not a string, leave it as is
+                builder.add(value);
+            }
+        }
+        return builder.build();
+    }
+
+    /**
+     * Add the children of the given node to itself.
+     * @param resource The resource whose children we're looking for
+     * @param resolver A reference to a resource resolver
+     * @param parentpath The path of the vocabulary to search through
+     * @return The input resource, with its children added as a property with key CHILDREN_PROPERTY
+     */
+    private JsonObject populateChildren(JsonObject resource, ResourceResolver resolver, String parentpath)
+    {
+        // Add child terms
+        Iterator<Resource> children = getVocabularyTermChildren(resource, resolver, parentpath);
+
+        // Copy the resource but add lfs:hasChildren to it
+        JsonObjectBuilder objectCopier = Json.createObjectBuilder();
+        for (String key : KEYS_TO_COPY) {
+            objectCopier.add(key, resource.get(key));
+        }
+        objectCopier.add(HAS_CHILDREN_PROPERTY, children.hasNext());
+
+        // Copy over our children to lfs:Children
+        JsonArrayBuilder childrenBuilder = Json.createArrayBuilder();
+        while (children.hasNext()) {
+            Resource child = children.next();
+
+            // Copying over the entire child bloats the response, so add only a subset of
+            // the child's properties
+            JsonObject childJson = child.adaptTo(JsonObject.class);
+            JsonObjectBuilder truncatedChild = Json.createObjectBuilder();
+            for (String key : KEYS_TO_COPY) {
+                truncatedChild.add(key, childJson.get(key));
+            }
+
+            childrenBuilder.add(truncatedChild.build());
+        }
+        objectCopier.add(CHILDREN_PROPERTY, childrenBuilder.build());
+
+        return objectCopier.build();
+    }
+
+    /**
+     * Determine if the given resource has children or not.
+     *
+     * @param resource The JsonObject to obtain the children of. Must have a child with key "id"
+     * @param resolver A reference to a ResourceResolver to use
+     * @param parentpath the location of the vocabulary whose children we're searching
+     * @return An iterator over the resource's children
+     */
+    private Iterator<Resource> getVocabularyTermChildren(JsonObject resource, ResourceResolver resolver,
```

Comment:
This class only deals with vocabulary terms, and the other methods don't use `VocabularyTerm` in their name, so for consistency and simplicity:

```suggestion
    private Iterator<Resource> getChildren(JsonObject resource, ResourceResolver resolver,
```